code/modules/power/singularity/collector.dm

```diff
@@ -113,18 +113,28 @@
 		loaded_tank = W
 		update_icon()
 	else if(W.GetID())
-		if(allowed(user))
-			if(active)
-				locked = !locked
-				to_chat(user, "<span class='notice'>You [locked ? "lock" : "unlock"] the controls.</span>")
-			else
-				to_chat(user, "<span class='warning'>The controls can only be locked when \the [src] is active!</span>")
-		else
-			to_chat(user, "<span class='danger'>Access denied.</span>")
-			return TRUE
+		togglelock(user)
+		return TRUE
```

Comment:
Move the return in the togglelock proc and only return true if togglelock returns true